### PR TITLE
[MAINTENANCE] Clean up legacy checkpoint tests and components

### PR DESCRIPTION
**Pull request by cdkini**
Slowly trimming our test suite to get to a place where we can rewrite the important ones and completely remove the legacy checkpoint

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/checkpoint/test_checkpoint.py [hunk 1]
```diff
@@ -80,146 +73,6 @@ def batch_request_as_dict() -> Dict[str, str]:
     }
 
 
-@pytest.mark.filesystem
-def test_basic_checkpoint_config_validation(
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint.py:84]: I've deleted various tests within this file that focus on behavior that is specific to the legacy checkpoint - please take a look at the test names and confirm that I'm correct in deleting these!

We'll probably need to do another pass but I'm just cleaning up as I go.

File: great_expectations/validation_operators/__init__.py [hunk 2]
```diff
@@ -13,7 +13,6 @@
 from .validation_operators import (
     ActionListValidationOperator,
     ValidationOperator,
-    WarningAndFailureExpectationSuitesValidationOperator,
```
Comment (cdkini) [inline great_expectations/validation_operators/__init__.py:16]: Not used at all. We still have `ActionListValidationOperator` holding us back from deleting this module but we should be close to removing it

File: tests/checkpoint/test_checkpoint_config.py [hunk 3]
```diff
@@ -1,59 +0,0 @@
-import pytest
-
-from great_expectations.checkpoint import Checkpoint
-
-DATA_CONTEXT_ID = "00000000-0000-0000-0000-000000000001"
-
-
-@pytest.fixture
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint_config.py:8]: Checkpoint config doesn't exist (Pydantic model joins the config and runtime obj)

File: tests/checkpoint/test_checkpoint_run_anonymizer.py [hunk 4]
```diff
@@ -1,44 +0,0 @@
-import pytest
-
-from great_expectations.checkpoint import Checkpoint
-
-DATA_CONTEXT_ID = "00000000-0000-0000-0000-000000000001"
-
-
-@pytest.fixture
-def checkpoint(
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint_run_anonymizer.py:9]: Just a fixture in this file?

**Review by tyler-hoffman — APPROVED** [review]

**Pull request merged**